code/game/machinery/computer/ftl_weapons.dm

```diff
@@ -196,7 +194,10 @@
 
 			. = 1
 		if("toggle_shields")
-			SSstarmap.ftl_shieldgen.on = !SSstarmap.ftl_shieldgen.on
+			if(!SSstarmap.ftl_shieldgen.stat)
+				SSstarmap.ftl_shieldgen.stat = POWEROFF
+			else if(SSstarmap.ftl_shieldgen.stat)
+				SSstarmap.ftl_shieldgen.stat = 0
```

Comment:
this means turning the shieldgen off and on again fixes any problem. do `stat &= ~POWEROFF` to toggle off the flag

also `|= POWEROFF` to toggle it on